overlord/snapstate/catalogrefresh.go

```diff
@@ -34,7 +35,10 @@ import (
 	"github.com/snapcore/snapd/overlord/state"
 )
 
-var catalogRefreshDelay = 24 * time.Hour
+var (
+	catalogRefreshDelayBase      = 24 * time.Hour
+	catalogRefreshDelayWithDelta = 24*time.Hour + 1 + time.Duration(rand.Intn(int(3*time.Hour)))
```

Comment:
should we smear it -3h..3h  ?  or smear it over 6h if only forward?

Reply:
the first proposal doesn't make (in this impl), we would reduce the interval below 24h which probably we don't want

Reply:
time.Duration is a int64,  we need to use rand.Int63n ?